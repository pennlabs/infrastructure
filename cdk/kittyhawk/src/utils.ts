export type NonEmptyArray<T> = [T, ...T[]];

export function nonEmptyMap<T, U>(
  arr: NonEmptyArray<T>,
  callbackfn: (value: T) => U
): NonEmptyArray<U> {
  return arr.map(callbackfn) as any as NonEmptyArray<U>;
}
