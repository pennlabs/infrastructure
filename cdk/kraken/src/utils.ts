export const buildId = (id: string, suffix: string) =>
  suffix ? `${id}-${suffix}` : id;
export const buildName = (name: string, id: string) => `${name} ${id}`.trim();
