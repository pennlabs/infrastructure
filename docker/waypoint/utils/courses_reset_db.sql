BEGIN;

DO $$
DECLARE
    target_tables text[] := ARRAY[
        'accounts_accesstoken', 'accounts_refreshtoken', 'alert_adddropperiod',
        'alert_pcademanddistributionestimate', 'alert_registration',
        'alert_v2_notification_history', 'alert_v2_registration', 'auth_group',
        'auth_group_permissions', 'auth_permission', 'auth_user', 'auth_user_groups',
        'auth_user_user_permissions', 'courses_apikey', 'courses_apikey_privileges',
        'courses_apiprivilege', 'courses_attribute', 'courses_attribute_courses',
        'courses_building', 'courses_comment', 'courses_comment_downvotes',
        'courses_comment_upvotes', 'courses_course', 'courses_department',
        'courses_friendship', 'courses_instructor', 'courses_meeting',
        'courses_ngssrestriction', 'courses_prengssrequirement',
        'courses_prengssrequirement_courses', 'courses_prengssrequirement_departments',
        'courses_prengssrequirement_overrides', 'courses_prengssrestriction',
        'courses_room', 'courses_section', 'courses_section_associated_sections',
        'courses_section_instructors', 'courses_section_ngss_restrictions',
        'courses_section_pre_ngss_restrictions', 'courses_statusupdate',
        'courses_topic', 'courses_userprofile', 'degree_degree', 'degree_degree_rules',
        'degree_degreeplan', 'degree_degreeplan_degrees', 'degree_dockedcourse',
        'degree_doublecountrestriction', 'degree_fulfillment',
        'degree_fulfillment_rules', 'degree_pdpbetauser', 'degree_rule',
        'degree_satisfactionstatus', 'django_admin_log',
        'django_celery_beat_clockedschedule', 'django_celery_beat_crontabschedule',
        'django_celery_beat_intervalschedule', 'django_celery_beat_periodictask',
        'django_celery_beat_periodictasks', 'django_celery_beat_solarschedule',
        'django_celery_results_taskresult', 'django_content_type',
        'django_migrations', 'django_session', 'options_option',
        'plan_primaryschedule', 'plan_schedule', 'plan_schedule_sections',
        'review_review', 'review_reviewbit', 'shortener_url'
    ];
    r RECORD;
    current_table text;
BEGIN
    -- Drop constraints
    FOR r IN (SELECT DISTINCT tc.table_schema, tc.constraint_name, tc.table_name, tc.constraint_type
              FROM information_schema.table_constraints tc
              WHERE tc.table_schema = 'public'
              AND tc.table_name = ANY(target_tables)
              AND tc.constraint_type != 'CHECK')
    LOOP
        EXECUTE format('ALTER TABLE %I.%I DROP CONSTRAINT IF EXISTS %I CASCADE',
            r.table_schema, r.table_name, r.constraint_name);
    END LOOP;

    -- Drop indexes
    FOR r IN (SELECT schemaname, tablename, indexname 
              FROM pg_indexes 
              WHERE schemaname = 'public'
              AND tablename = ANY(target_tables))
    LOOP
        EXECUTE format('DROP INDEX IF EXISTS %I.%I CASCADE',
            r.schemaname, r.indexname);
    END LOOP;

    -- Truncate tables
    FOREACH current_table IN ARRAY target_tables
    LOOP
        IF EXISTS (SELECT FROM information_schema.tables 
                  WHERE table_schema = 'public' 
                  AND table_name = current_table) THEN
            EXECUTE format('TRUNCATE TABLE public.%I CASCADE', current_table);
        END IF;
    END LOOP;

    -- Reset sequences
    FOR r IN 
        SELECT sequence_name 
        FROM information_schema.sequences 
        WHERE sequence_schema = 'public'
    LOOP
        EXECUTE format('ALTER SEQUENCE public.%I RESTART WITH 1', r.sequence_name);
    END LOOP;
END $$;

COMMIT;