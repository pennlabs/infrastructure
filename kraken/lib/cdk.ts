import { CheckoutJob, Workflow, Stack, WorkflowProps } from 'cdkactions';
import { Construct } from 'constructs';
import * as dedent from 'dedent-js';

export interface CDKPublishProps {
  defaultBranch?: string;
  nodeVersion?: string;
}

export class CDKStack extends Stack {
  public constructor(scope: Construct, id: string, config?: CDKPublishProps, overrides?: Partial<WorkflowProps>) {
    // Build config
    const fullConfig: Required<CDKPublishProps> = {
      defaultBranch: 'master',
      nodeVersion: '14',
      ...config,
    };
    super(scope, id);
    const workflow = new Workflow(this, id, {
      name: `Publish ${id}`,
      on: {
        push: {
          paths: [`${id}/**`],
        },
      },
      ...overrides,
    });
    new CheckoutJob(workflow, 'publish', {
      runsOn: 'ubuntu-latest',
      steps: [{
        name: 'Cache',
        uses: 'actions/cache@v2',
        with: {
          path: '**/node_modules',
          // TODO: fix extra quotes here
          key: `v0-\${{ hashFiles('${id}/yarn.lock') }}`,
        },
      },
      {
        name: 'Install Dependencies',
        run: dedent`cd ${id}
        yarn install --frozen-lockfile`,
      },
      {
        name: 'Test',
        run: dedent`cd ${id}
        yarn test`,
      },
      {
        name: 'Upload Code Coverage',
        run: dedent`ROOT=$(pwd)
        cd ${id}
        yarn run codecov -p $ROOT -F ${id}`,
      },
      {
        name: 'Publish to npm',
        run: dedent`cd ${id}
        npm login --always-auth
        npm publish --access public`,
        if: `github.ref == 'refs/heads/${fullConfig.defaultBranch}'`,
      }],
      // TODO: publish docs
      container: {
        image: `node:${fullConfig.nodeVersion}`,
      },
    });
  }
}
