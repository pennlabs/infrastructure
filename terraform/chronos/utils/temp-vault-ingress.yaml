apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: temp-vault-ingress
  namespace: default
spec:
  rules:
    - host: vault.upenn.club
      http:
        paths:
          - backend:
              serviceName: vault
              servicePort: 8200
            path: /
