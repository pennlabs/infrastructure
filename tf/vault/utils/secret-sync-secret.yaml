apiVersion: v1
kind: Secret
metadata:
  name: secret-sync
  annotations:
    vault-secret: secrets/chronos/default/secret-sync
type: Opaque
stringData:
  ROLE_ID: $ROLE_ID
  SECRET_ID: $SECRET_ID
