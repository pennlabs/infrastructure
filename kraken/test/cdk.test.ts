import * as fs from 'fs';
import { CDKStack } from '../lib';
import { TestingApp } from './utils';

test('default', () => {
  const app = TestingApp({ createValidateWorkflow: false });
  new CDKStack(app, 'example');
  app.synth();
  expect(fs.readdirSync(app.outdir)).toEqual([
    'cdkactions_example.yaml',
  ]);
  expect(fs.readFileSync(`${app.outdir}/cdkactions_example.yaml`, 'utf-8')).toMatchSnapshot();
},
);
