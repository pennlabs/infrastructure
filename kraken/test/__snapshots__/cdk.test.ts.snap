// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default 1`] = `
"# Generated by cdkactions. Do not modify
# Generated as part of the 'example' stack.
name: Publish example
on:
  push:
    paths:
      - example/**
jobs:
  publish:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Cache
        uses: actions/cache@v2
        with:
          path: '**/node_modules'
          key: 'v0-\${{ hashFiles('example/yarn.lock') }}'
      - name: Install Dependencies
        run: |-
          cd example
          yarn install --frozen-lockfile
      - name: Test
        run: |-
          cd example
          yarn test
      - name: Upload Code Coverage
        run: |-
          ROOT=$(pwd)
          cd example
          yarn run codecov -p $ROOT -F example
      - name: Publish to npm
        run: |-
          cd example
          npm login --always-auth
          npm publish --access public
        if: github.ref == 'refs/heads/master'
        env:
          NPM_AUTH_TOKEN: \${{ secrets.NPM_AUTH_TOKEN }}
    container:
      image: 'node:14'
"
`;
